daycare/test_daycare.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+from .models import DayCare
+from django.core.exceptions import ValidationError
+from django.contrib.auth.models import User
+
+EMAIL = "test@gmail.com"
+USERNAME = "testuser01"
+PASSWORD = "pass"
+NAME = "Puppies"
+DESCRIPTION = "This is the first daycare test"
+PRICE_PER_DAY = 10
+CAPACITY = 50
+AREA = "Merkaz"
+CITY = "Tel-Aviv"
+ADDRESS = "The best street 5"
+
+
+@pytest.fixture
+def create_daycare_user():
+    return DayCare.create(email=EMAIL, username=USERNAME, password=PASSWORD,
+                          name=NAME, description=DESCRIPTION,
+                          price_per_day=PRICE_PER_DAY,
+                          capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+
+@pytest.mark.django_db()
+class TestDaycareModel:
+    def test_persist_daycare(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+    def test_del_daycare(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+        create_daycare_user.delete()
+        assert create_daycare_user not in DayCare.objects.all()
+        assert create_daycare_user not in User.objects.all()
+
+    def test_daycare_user_delete(self, create_daycare_user):
+        assert create_daycare_user.user in User.objects.all()
+        assert create_daycare_user in DayCare.objects.all()
+
+        create_daycare_user.user.delete()
+        assert create_daycare_user not in DayCare.objects.all()
+
+    def test_daycare_user_creation_with_invalid_email(self):
+        with pytest.raises(ValidationError, match="'Enter a valid email address.'"):
+            DayCare.create(username='testuser02', email="invalid mail format", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+    def test_daycare_user_creation_with_same_user_name(self):
+        with pytest.raises(ValidationError, match="Invalid username"):
+            DayCare.create(email=EMAIL, username="testuser01", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+            DayCare.create(email=EMAIL, username="testuser01", password=PASSWORD,
+                           name=NAME, description=DESCRIPTION,
+                           price_per_day=PRICE_PER_DAY,
+                           capacity=CAPACITY, area=AREA, city=CITY, address=ADDRESS)
+
+    def test_daycare_user_with_invalid_name(self):
```

Comment:
I used the clear() methods because the field that I'm testing is build-in Django validations
you can take a look [here](https://docs.djangoproject.com/en/4.0/ref/models/instances/#validating-objects) for more information.
And this is the reason why I didn't use the `match` option and used `Exception` because there is no "match" for custom validate function, so I wrote my own text in the exception option.

Reply:
Well then clear() should be called by create(), there is no point in creating an object if its not valid.
In general validation should be part of the creation flow, along with persistence.

As for the match thing, the match arg was meant for catching exception by pattern
https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions

Reply:
Fixed

Reply:
It doesn't seem that new changes were pushed.

Reply:
By bad, now its done @ormergi